atlasdb-config/src/main/java/com/palantir/atlasdb/config/TimestampClientConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.atlasdb.AtlasDbConstants;
+
+@JsonSerialize(as = ImmutableTimestampClientConfig.class)
+@JsonDeserialize(as = ImmutableTimestampClientConfig.class)
+@Value.Immutable
+public abstract class TimestampClientConfig {
+    @Value.Default
+    public boolean enableTimestampBatching() {
+        return AtlasDbConstants.DEFAULT_ENABLE_TIMESTAMP_BATCHING;
+    }
+
+    // TODO (jkong): Make timestamp wait intervals configurable.
+    // This should ONLY be done once the timestamp client supports nanosecond precision;
+    // millisecond precision isn't too useful (realistically it's very unlikely you want to set this beyond
+    // 2 ms or so).
```

Comment:
Yeah, as you mention I think the millisecond level batching periods here are pretty useless. What we probably want instead is the ability to specify a maximum number of parallel requests.

If we know the point (i.e., number of concurrent threads) at which the benefits of batching outweigh the extra latency, we can configure services with a sane maximum parallelism.